pkg/dispatcher/dispatcher.go

```diff
@@ -100,14 +100,16 @@ func (p *Dispatcher) dispatch(event events.SchedulingEvent) error {
 	if !p.isRunning() {
 		return fmt.Errorf("dispatcher is not running")
 	}
-
-	select {
-	case p.eventChan <- event:
-		return nil
-	default:
-		return fmt.Errorf("failed to dispatch event," +
-			" event channel is full or closed")
+	for p.isRunning() {
+		select {
+		case p.eventChan <- event:
+			return nil
+		case <-time.After(time.Second * 3):
```

Comment:
Should we run this in a goroutine?
if we can enqueue the event, then go for it,
otherwise, we launch a goroutine that tries to enqueue the event in every 3s, and try at most 5 minutes to die.

Reply:
We should not let it sit there and wait. If `dispatch` becomes blocking `Dispatch` will also become a blocking call. That will have a flow on effect and could block the whole system.

We need to be very careful about this.